packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -218,6 +232,19 @@ function reducer<T, C>(data: AsyncListState<T, C>, action: Action<T, C>): AsyncL
             items: action.type === 'loading' ? [] : data.items,
             abortController: action.abortController
           };
+        case 'loadingMore':
+          // If already loading more and another loading more is triggered, abort the new load more.
+          // Do not overwrite the data.abortController
+          action.abortController.abort();
```

Comment:
How would this case occur? Don't we prevent it here?

https://github.com/adobe/react-spectrum/blob/48c95b9ec06b044841c031e43ffcff25dad54028/packages/%40react-stately/data/src/useAsyncList.ts#L327-L329

Reply:
That is what I thought at first, but if `loadMore` is called multiple times in quick succession it is apparently possible that `data.state` doesn't update in time. I only noticed this in the async combobox story where Virtualizer occasionally called `loadMore` multiple times when the menu and break cuz loadingMore wasn't a valid action when state was loadingMore

 I added a test specifically for this here: https://github.com/adobe/react-spectrum/blob/48c95b9ec06b044841c031e43ffcff25dad54028/packages/%40react-stately/data/test/useAsyncList.test.js#L746-L777

Reply:
If you reset this branch to `121f4bc` (aka before I made these changes) and reuse the above test with a try catch around the double `.loadMore` calls you can see the error `Invalid action "loadingMore" in state "loadingMore"`

Reply:
Oh hmm, that makes sense. I guess we could solve that with a ref to track `isLoading`, which would be updated synchronously by `dispatchFetch`. Possibly better than starting two requests and then immediately aborting one?

Reply:
oh that makes sense. Ideally we wouldn't start a request only to immediately abort it though. One option would be to track the loading state in a ref so we don't have to wait for render to get the updated value. Another option would be to only trigger the request from within `reduce`, but this may delay the start for longer than necessary.

Reply:
kk, implemented this kinda. Added a separate `loadingState` ref that gets updated in `dispatchFetch` but preserved `data.state`. Was the intention here to completely remove `data.state` and replace it with the `loadingState` ref?